### PR TITLE
PIM-7420: Keep item ordering in item picker

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

Items in the item picker (only used today for the assets in EE) should be organized in the order they are selected.

However, this is not working if the codes of the entities are integers and not strings. As the fetcher creates an object using the item codes as key, when those codes are integers, the items are reorganized, automatically sorted by code ascending, even if the items were already sorted by the back-end.

So the only way is to reorder the items according to the original list of codes provided to the fetcher.

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js [hunk 1]
```diff
@@ -310,12 +310,16 @@ define(
...
-                FetcherRegistry.getFetcher(this.config.fetcher).fetchByIdentifiers(this.getItems())
-                    .then(function (items) {
-                        this.renderBasket(items);
-                        this.delegateEvents();
-                    }.bind(this));
+            updateBasket: function (itemCodes) {
+                FetcherRegistry.getFetcher(this.config.fetcher).fetchByIdentifiers(this.getItems()).then(items => {
+                    let orderedItems = [];
+                    itemCodes.forEach(itemCode => {
+                        orderedItems = orderedItems.concat(items.filter(item => item.code === itemCode));
```
Comment (rouliane) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: i'm not good in JS, but i'm not sure what the line does... It erase orderedItems in each iteration ?

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: it is the same than an array_merge in PHP

Comment (fitn) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: Could you explain to me what this code is used for?
In the doc, it's written fetch the new items and render the basket. To me, if it is about adding new items, it should be a push and not a merge.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: Indeed, a `push` would be way better. I forgot it exists.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: A push cannot work, as I use an `array.filter` which will return an array.
Or I need to perform an `array.push(resultOfFilter[0])`.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: And for the explanation: it does what I told in the description. The `orderedItems` are a new list of the elements of `items`, but ordered by their codes (meaning `item.code`) as those are in `itemCodes`.

Comment (anaelChardan) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: Maybe you could do ```orderedItems.flatMap(itemCode => items.filter(item => item.code === idemCode))``` which is a little bit more functional and is immutable.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: mh. MDN lists it as an experimental technology. And for now, only Chrome supports it, and Firefox only on nightly builds, so not an option.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: What underscore method are you thinking of? If there is one that can shorten this writing, I am in for, but if it is just syntax sugar for the exact same operations, I prefer keep native javascript.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: Then it will come, one day :-)

Comment (fitn) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: Please, as discussed yesterday together, add a few comments to explain why we are doing that.

Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/common/item-picker.js]: Done

**Pull request merged**